angular/src/directives/control-value-accessors/value-accessor.ts

```diff
@@ -110,13 +110,17 @@ export class ValueAccessor implements ControlValueAccessor, AfterViewInit, OnDes
 
 export const setIonicClasses = (element: ElementRef): void => {
   raf(() => {
-    const input = element.nativeElement as HTMLElement;
+    const input = element.nativeElement as HTMLInputElement;
+    const hasValue = typeof input.value !== 'undefined' && input.value.toString().length > 0;
```

Comment:
We need a null check here as the `value` prop is nullable: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.tsx#L194

Reply:
We should probably also write tests for when the value is `undefined` and `null` so this does not regress

Reply:
The internals of `ion-input` don't check for `null`: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.tsx#L327-L330

While you can assign `null` as a value, the evaluated value will always be a string (`''` for undefined/null). This aligns with how the HTML input element functions.

Do we still need to cover those cases?

Reply:
From what it looks like `getValue` is only used when parsing the value internally:
Usage 1: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.tsx#L236
Usage 2: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.tsx#L426
Usage 3: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/input/input.tsx#L431

None of these usages update the `value` property to be the empty string, and your code is checking the `value` prop, not calling `getValue`.

I think it's worth testing just to be sure. If we convert `null` to the empty string elsewhere such that this is not a concern, then we can proceed without the `null` check. At the moment it is not clear if that is the case.